lib/usd/translators/shading/usdFileTextureWriter.cpp

```diff
@@ -452,6 +472,178 @@ void PxrUsdTranslators_FileTextureWriter::Write(const UsdTimeCode& usdTime)
         const TfToken wrapT = wrapV ? _tokens->repeat : _tokens->black;
         shaderSchema.CreateInput(_tokens->wrapT, SdfValueTypeNames->Token).Set(wrapT, usdTime);
     }
+
+    WriteTransform2dNode(usdTime, shaderSchema);
+}
+
+void PxrUsdTranslators_FileTextureWriter::WriteTransform2dNode(
+    const UsdTimeCode&    usdTime,
+    const UsdShadeShader& texShaderSchema)
+{
+    MStatus status;
+
+    const MFnDependencyNode depNodeFn(GetMayaObject(), &status);
+    if (status != MS::kSuccess) {
+        return;
+    }
+
+    // Gather UV transform data.  If it differs from default values,
+    // create a Transform2d Node, connect it to the output "result"
+    // of the TexCoordReader node and the input "st" of the FileTexture node.
+    const GfVec2f translationFallbackValue(0.0f, 0.0f);
+    const float   rotationFallbackValue = 0.0f;
+    const GfVec2f scaleFallbackValue(1.0f, 1.0f);
+
+    GfVec2f translationValue = translationFallbackValue;
+    float   rotationValue = rotationFallbackValue;
+    GfVec2f scaleValue = scaleFallbackValue;
+
+    bool transformationsAreAuthored = false;
+
+    // Translation
+    const MPlug offsetPlug = depNodeFn.findPlug(
+        _tokens->offset.GetText(),
+        /* wantNetworkedPlug = */ true,
+        &status);
+    if (status != MS::kSuccess) {
+        return;
+    }
+
+    if (UsdMayaUtil::IsAuthored(offsetPlug)) {
+        for (size_t i = 0u; i < GfVec2f::dimension; ++i) {
+            translationValue[i] = offsetPlug.child(i).asFloat(&status);
+            if (status != MS::kSuccess) {
+                return;
+            }
+        }
+
+        if (translationValue != translationFallbackValue) {
+            transformationsAreAuthored = true;
+        }
+    }
+
+    // Rotation
+    const MPlug rotateUVPlug = depNodeFn.findPlug(
+        _tokens->rotateUV.GetText(),
+        /* wantNetworkedPlug = */ true,
+        &status);
+    if (status != MS::kSuccess) {
+        return;
+    }
+
+    if (UsdMayaUtil::IsAuthored(rotateUVPlug)) {
+        rotationValue = rotateUVPlug.asFloat(&status);
+        if (status != MS::kSuccess) {
+            return;
+        }
+
+        if (rotationValue != rotationFallbackValue) {
+            transformationsAreAuthored |= true;
+        }
+
+        // Convert rotation from radians to degrees
+        rotationValue = ((rotationValue / M_PI) * 180.0f);
```

Comment:
`pxr/base/gf/math.h` Provides function for this:

```C++
rotationValue = GfRadiansToDegrees(rotationValue)
```

Reply:
Thanks, swapped that out.